modules/pam_faillock/faillock.c

```diff
@@ -74,6 +74,9 @@ open_tally (const char *dir, const char *user, uid_t uid, int create)
 
 	if (create) {
 		flags |= O_CREAT;
+		if (access(dir, F_OK) != 0) {
+			mkdir(dir, 0755);
+		}
```

Comment:
Do we really have to test the path before mkdir invocation?

Reply:
I prefer keeping the test there. The code is racy anyway even without it and the added access call should not cause any performance issue.